hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/ContractUpdateHandler.java

```diff
@@ -74,6 +107,124 @@ private boolean hasCryptoAdminKey(final ContractUpdateTransactionBody op) {
 
     @Override
     public void handle(@NonNull final HandleContext context) throws HandleException {
-        throw new UnsupportedOperationException("Not implemented");
+        final var txn = requireNonNull(context).body();
+        final var op = txn.contractUpdateInstanceOrThrow();
+        final var target = op.contractIDOrThrow();
+
+        final var accountStore = context.readableStore(ReadableAccountStore.class);
+
```

Comment:
In token service we have a method called `validateSemantics` in each handle to do any validation we want to do, based on state/config. We can add such method and do the validations happening in `customizerFor` in mono-service imo

Reply:
The method was called `validate` instead of `validateSemantics`. I renamed it to `validateSemantics` and pulled it from the `update` method. Now it should be more clear